lib/apaka/packaging/autoproj2adaptor.rb

```diff
@@ -545,15 +545,22 @@ def dependencies(pkg, with_rock_release_prefix = true)
                     # Convert native ruby gems package names to rock-xxx
                     if pkg_handler == "gem"
                         pkg_list.each do |name|
-                            version = nil
-                            if name =~ /([<>=]=?.*)$/
-                                version = $1
-                            end
+                            if name.class == Hash
+                                git = name['git']
+                                name = name['name']
+                                version = nil
+                            else
+                                git = nil
+                                version = nil
+                                if name =~ /([<>=]=?.*)$/
+                                    version = $1
+                                end
 
-                            name = name.gsub(/[<>=]=?.*$/,"")
+                                name = name.gsub(/[<>=]=?.*$/,"")
+                            end
 
-                            extra_gems << [name, version]
-                            non_native_dependencies << [name, version]
+                            extra_gems << [name, version, git]
```

Comment:
Where are the changes to the rest of the code in order` account for this additional field, e.g.,
https://github.com/rock-core/tools-apaka/blob/f329217331b01baa90e02ea3b0b80a21ea54e9af/lib/apaka/cli/package.rb#L91